### PR TITLE
Updated PHP CS Fixer to 2.0

### PR DESCRIPTION
**Pull request by javiereguiluz**
This finishes #407.

### CODE REVIEW

File: .php_cs [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
     ->exclude('var')
     ->exclude('vendor')
-    ->exclude('web/bundles')
-    ->exclude('web/css')
-    ->exclude('web/fonts')
-    ->exclude('web/js')
     ->notPath('web/config.php')
-    ->in(__DIR__)
+    // Doctrine fixtures are excluded to avoid these PHP-CS-Fixer errors:
+    // "Files were not fixed due to errors reported during linting before fixing"
```
Comment (keradus) [inline .php_cs]: why doctrine fixtures are invalid PHP code ?

File: .php_cs [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
+    ->setRules([
+        '@Symfony' => true,
+        '@Symfony:risky' => true,
+        'array_syntax' => ['syntax' => 'short'],
+        'no_useless_else' => true,
+        'no_useless_return' => true,
+        'ordered_imports' => true,
+        'phpdoc_order' => true,
+        'php_unit_strict' => true,
+        'strict_comparison' => true,
```
Comment (keradus) [inline .php_cs]: I am totally for this rule, but it's not from Symfony ruleset. Why include it in demo ? That would mean anyone would include here anything with time passing, right ? I would either keep Symfony rulesets only or propose those changes to be included in the ruleset in the first place.

Problematic parts are rules like `ordered_imports`, which Symfony wants to follow in new code, but don't want to apply it for whole codebase to avoid merging conflicts

File: composer.lock [hunk 3]
```diff
@@ -73,7 +73,7 @@
                 "docblock",
                 "parser"
             ],
-            "time": "2015-08-31 12:32:49"
+            "time": "2015-08-31T12:32:49+00:00"
```
Comment (keradus) [inline composer.lock]: btw, is the lock file really needed for demo project ?

**Comment by javiereguiluz** [comment]
@keradus in 0aa68ef I've removed everything except the default Symfony rules. Is that what you were proposing?

File: composer.lock (hunk 3 above)
Comment (stof) [inline composer.lock]: I would say yes, as it showcases best practices for projects, and it ensures that people cloning the demo gets working dependencies, which is a better experience for a first try

File: .php_cs [hunk 4]
```diff
@@ -1,29 +1,22 @@
...
-    ->exclude('app/data')
-    ->exclude('app/Resources')
+$finder = PhpCsFixer\Finder::create()
+    ->in(__DIR__)
     ->exclude('var')
     ->exclude('vendor')
-    ->exclude('web/bundles')
-    ->exclude('web/css')
-    ->exclude('web/fonts')
-    ->exclude('web/js')
```
Comment (stof) [inline .php_cs]: I would keep excluding them (no need to look for them to discover they are unsupported, while we can skip entering these folders entirely)

File: .php_cs (hunk 1 above)
Comment (stof) [inline .php_cs]: IMO, we must indeed rather fix the issue, which is a bad git conflict resolution: https://github.com/symfony/symfony-demo/blob/eba7c9f21bbd2dca312c6f767b77aafc60d47d33/src/AppBundle/DataFixtures/ORM/PostFixtures.php#L80

File: .php_cs [hunk 5]
```diff
@@ -1,29 +1,22 @@
...
-$finder = Symfony\CS\Finder::create()
-    ->ignoreDotFiles(true)
-    ->ignoreVCS(true)
-    ->exclude('app/config')
-    ->exclude('app/data')
-    ->exclude('app/Resources')
+$finder = PhpCsFixer\Finder::create()
+    ->in(__DIR__)
     ->exclude('var')
     ->exclude('vendor')
```
Comment (stof) [inline .php_cs]: this is already excluded by ``PhpCsFixer\Finder::create()``

File: composer.lock (hunk 3 above)
Comment (keradus) [inline composer.lock]: the issue with it that usually the user won't update it, which mean he will use outdated once

**Comment by keradus** [comment]
> @keradus in 0aa68ef I've removed everything except the default Symfony rules. Is that what you were proposing?

no, i proposed to evaluate the rules and maybe include them into `@Symfony` ruleset itself.
or, if Symfony project will refuse to apply the rules on codebase but expect them for new code, create sth like `@SymfonyWishList` (with better name) to track that kind of rules that should apply for new code only.

File: src/AppBundle/DataFixtures/ORM/PostFixtures.php [hunk 6]
```diff
@@ -107,8 +106,6 @@ private function addRandomTags(Post $post)
         }
     }
 
-=======
->>>>>>> pull/431:src/AppBundle/DataFixtures/ORM/LoadFixtures.php
```
Comment (keradus) [inline src/AppBundle/DataFixtures/ORM/PostFixtures.php]: btw, this conflict could be resolved in standalone PR ;)
to keep PRs atomic and clean

File: .php_cs (hunk 4 above)
Comment (keradus) [inline .php_cs]: also, sadly, I met few JS packages that are distributed with... **PHP** files (as a demo usage)

File: composer.lock (hunk 3 above)
Comment (javiereguiluz) [inline composer.lock]: @keradus you are right, butt keep in mind that this demo application is a bit special. First, we download it as a ZIP file with all the dependencies installed ... so 99% of users never use Composer with this app. Second, the demo app is designed as a temporary utility (to learn Symfony, to showcase some feature, to do a quick benchmark) so you usually install the demo and delete it pretty quickly ... and then download it again. So 99% of people won't get old dependencies or don't care about them.

Comment (stof) [inline composer.lock]: thus, if people want an updated version of the demo, they should not only update the vendors, but also the demo itself (and so they would get our new lock file and run ``composer install``)

File: .php_cs (hunk 2 above)
Comment (stof) [inline .php_cs]: well, Symfony itself decided to order new use statements only, due to maintenance constraints. But this does not forbid this demo project to be strict about them.
The same is true for the short array syntax. This demo decided to use it.

regarding ``no_useless_else``, if it does what I think it does according to its name, it should be part of the ruleset, as Symfony indeed mandates that we don't use ``else`` when the ``if`` quits the execution path (through ``return``, ``throw``, ``continue`` or ``break``

File: composer.lock (hunk 3 above)
Comment (keradus) [inline composer.lock]: ok, thus :+1:
(on a side not, that approach would require this demo project to update lock file on regular basis)

**Comment by keradus** [comment]
great to see this happen ;)
